src/ui/ContextMenu/items/OperatorMenuItem.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { ReactElement, ReactNode, useCallback, useRef, useState } from 'react';
+import { BaseChannel, Role, User } from '@sendbird/chat';
+import { MenuItem } from '..';
+import { Member } from '@sendbird/chat/groupChannel';
+import { OpenChannel } from '@sendbird/chat/openChannel';
+
+export interface OperatorMenuItemProps {
+  channel: BaseChannel;
+  user: User & { isMuted: boolean };
+  className?: string | Array<string>;
+  children: ReactElement | ReactElement[] | ReactNode;
+  disable?: boolean;
+  dataSbId?: string;
+  onChange?: (channel: BaseChannel, user: User, isOperator: boolean) => void;
+  onError?: (reason: any) => void;
+}
+
+export const OperatorMenuItem = ({
+  channel,
+  user,
+  className = '',
+  children,
+  disable = false,
+  dataSbId = '',
+  onChange = () => {},
+  onError = () => {},
+}: OperatorMenuItemProps): ReactElement => {
+  const [isOperator, setIsOperator] = useState(channel instanceof OpenChannel
+    ? channel.isOperator(user)
+    : (user as Member).role === Role.OPERATOR);
+  const isProcessing = useRef(false);
+
+  const onClickHandler = useCallback(() => {
+    if (!isProcessing.current) {
+      isProcessing.current = true;
+      if (isOperator) {
+        channel.removeOperators([user.userId])
+          .then(() => {
+            setIsOperator(false);
+            onChange(channel, user, false);
+            isProcessing.current = false;
+          })
+          .catch(err => {
+            onError(err);
+            isProcessing.current = false;
+          });
+      } else {
+        channel.addOperators([user.userId])
+          .then(() => {
+            setIsOperator(true);
+            onChange(channel, user, true);
+            isProcessing.current = false;
+          })
+          .catch(err => {
+            onError(err);
+            isProcessing.current = false;
+          });
+      }
+    }
+  }, [isProcessing.current]);
```

Comment:
Since `useRef` also returns an object, it ensures the latest value without dependencies in hooks such as `useEffect` or `useCallback`, where you need to reflect the latest value through dependencies.

```suggestion
  }, []);
```

And since the click handler business logic changes based on the value of `isOperator`, it seems like we should add it to the dependency array!

```suggestion
  }, [isOperator]);
```

Reply:
I added the `isProcessing` to the dependencies that it's the only reference that should change the callback. `isOperator` is a value that we don't have to observe for callback refresh.

Reply:
Added the deps